programs/monaco_protocol/src/instructions/math.rs

```diff
@@ -25,10 +27,13 @@ pub fn calculate_for_payout(stake: u64, price: f64) -> u64 {
     Decimal::from(stake).mul(price_decimal).to_u64().unwrap()
 }
 
-pub fn stake_precision_is_within_range(stake: u64, decimal_limit: u8) -> bool {
-    Decimal::new(stake as i64, decimal_limit as u32)
-        .fract()
-        .is_zero()
+pub fn stake_precision_is_within_range(stake: u64, decimal_limit: u8) -> Result<bool> {
+    let mut stake_decimal = Decimal::from_u64(stake).unwrap();
+    require!(
+        stake_decimal.set_scale(decimal_limit as u32).is_ok(),
+        CoreError::ArithmeticError
+    );
```

Comment:
my opinion is that if you doing `unwrap` on line 31 it kinda defeats the proper error throw on next line

Reply:
`from_u64` only returns an `Option` so that there is conformity in return types. It is not possible for `None` to be returned. 

The issue with `set_scale` is that it returns a `Result<(), Error>` as it checks parameter value ensuring it is not greater than an internal `MAX_PRECISION`. 

Barring your humorous suggestion below, I'm not sure how better to deal with the fact that `set_scale` might error.

Reply:
thsi si probably spliting hair but you coudl just unwrap it as the above
`stake_decimal.set_scale(decimal_limit as u32).unwrap();`
it will panic on error

Reply:
i'll just approve

Reply:
I don't think I share you opinion on this one. I'd rather not have actual protocol code that can panic if we can help it. 

`decimal_limit` comes from the market whereas `stake_precision_is_within_range` is called during order creation.
Assuming a token can be created with decimals count greater than `MAX_PRECISION`  and used for a market, then order creation would always panic for such a market. 

I think we can tidy this up further but not necessary for this fix IMO. I'll create an issue so a proper error can be returned on order creation failure should such a market be created in the future. Or implement a max decimals limit for market creation.